src/main/java/codec/Reader.java

```diff
@@ -0,0 +1,48 @@
+package codec;
+
+import java.nio.ByteBuffer;
+import java.nio.channels.SocketChannel;
+
+public class Reader {
+    public static final byte MAX_MESSAGE_LENGTH_BYTE = 4;
+    private final SocketChannel socketChannel;
+
+    public Reader(SocketChannel socketChannel) {
+        this.socketChannel = socketChannel;
+    }
+
+    public int getLengthMessage() {
+        ByteBuffer messageSize = ByteBuffer.allocate(MAX_MESSAGE_LENGTH_BYTE);
+        try {
+            if (socketChannel.read(messageSize) > -1) {
```

Comment:
We might read less than `MAX_MESSAGE_LENGTH_BYTE`. The `read` method returns the actual number of bytes that were read